app/components/AudioPlayer/index.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useRef } from 'react';
+import { PlayCircleFilled, PauseCircleFilled } from '@ant-design/icons';
+import { Tooltip } from 'antd';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const PlayIcon = styled(PlayCircleFilled)`
+  && {
+    font-size: 250%;
+  }
+`;
+const PauseIcon = styled(PauseCircleFilled)`
+  && {
+    font-size: 250%;
+  }
+`;
+const Timeline = styled.div`
+  && {
+    position: relative;
+    width: 100%;
+    height: 100%;
+    background-color: #e3e3e3;
+    border: 1px solid black;
+  }
+`;
+const TimelineContainer = styled.div`
+  && {
+    position: relative;
+    width: 100%;
+    margin: 1em 5px;
+    height: 5px;
+  }
+`;
+
+const Progress = styled.div`
+  && {
+    position: absolute;
+    left: 0;
+    top: 0;
+    height: 100%;
+    background-color: coral;
+  }
+`;
+const Controls = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    margin-top: 1em;
+  }
+`;
+
+const AudioPlayer = ({ source }) => {
+  const [playing, setPlaying] = useState(false);
+  const audio = useRef(null);
+  const timeline = useRef(null);
+  const progress = useRef(null);
+  const controls = useRef(null);
+  const currentTime = useRef(null);
+
+  const calculatePercentPlayed = () => {
+    let percentPlayed = (audio.current.currentTime / audio.current.duration).toFixed(2) * 100;
+    progress.current.style.width = `${percentPlayed}%`;
+  };
+  const calculateCurrentValue = currentTime => {
+    const currentMinute = parseInt(currentTime / 60) % 60;
+    const currentSecondsLong = currentTime % 60;
+    const currentSeconds = currentSecondsLong.toFixed();
+    const currentTimeFormatted = `${currentMinute < 10 ? `0${currentMinute}` : currentMinute}:${
+      currentSeconds < 10 ? `0${currentSeconds}` : currentSeconds
+    }`;
+
+    return currentTimeFormatted;
+  };
+  const initProgressBar = () => {
+    const media = audio.current;
+    const time = calculateCurrentValue(media.currentTime);
+
+    function seek(e) {
+      const percent = e.offsetX / timeline.current.offsetWidth;
+      media.currentTime = percent * media.duration;
+    }
+
+    currentTime.current.innerHTML = time;
+    timeline.current.addEventListener('click', seek);
+    calculatePercentPlayed();
+  };
+
+  const onEnded = () => {
+    progress.current.style.width = 0;
+    currentTime.current.innerHTML = '00:00';
+
+    setPlaying(false);
+  };
+
+  const play = () => {
+    setPlaying(true);
+    audio.current.play();
+  };
+
+  const pause = () => {
+    setPlaying(false);
+    audio.current.pause();
+  };
+
+  return (
+    <div>
+      <Controls ref={controls}>
+        {playing ? (
+          <Tooltip title="pause">
```

Comment:
use react-intl. 

you can directly use the `translate` function

Reply:
got it!